### PR TITLE
Introduce current_path and current_url

### LINKED ISSUE #2002: Introduce current_path and current_url
Today those need to be built by hand. See [discussion here](https://elixirforum.com/t/how-to-get-the-current-url/2390/6).

### PR DESCRIPTION
**Pull request by chrismccord**
@josevalim I wasn't sure the best place for these to live. My first goto was the Router.Helpers, but there is no need for us to inject the functions, and the module itself is private (moduledoc false). They are also more discoverable on Phoenix.Controller, but we'll have to import them in web.ex/view, or have folks import as needed. Wdyt?

### CODE REVIEW

**Comment by josevalim** [comment]
It looks great to me. My only question is if we should expose the version
that accepts the endpoint. I can't think of a use case for it, so maybe we
should make that clause private?

**Comment by chrismccord** [comment]
I also went back and forth on the need for the endpoint variant. I'm fine killing it, but the reason I added it was this. Imagine we don't have it, and someone who needs it will do:

    Endpoint.url() <> current_path(conn)

There no problem, unless the request path is "/" without query params, then they will end up with `"example.com/"` with a trailing slash, which is something our `current_url` handles. I could go either way, and it's always easy for us to add later if people are needing it. I'll let you decide to kill it with :fire: or not :)

**Comment by josevalim** [comment]
@chrismccord I believe we should always include the "/". It is a required part of URLs. So we can nuke the endpoint one if that is the reason of its existence. Btw, why are we not including the scheme? Such as http or https, similar to how our router helpers would do?

File: lib/phoenix/controller.ex [hunk 1]
```diff
@@ -1092,6 +1092,73 @@ defmodule Phoenix.Controller do
...
+    current_url(conn, endpoint_module(conn), params)
+  end
+  def current_url(%Plug.Conn{} = conn, endpoint) when is_atom(endpoint) do
+    Path.join(endpoint.url(), current_path(conn))
+  end
+  def current_url(%Plug.Conn{} = conn, endpoint, %{} = params) when is_atom(endpoint) do
+    Path.join(endpoint.url(), current_path(conn, params))
+  end
+
+
```
Comment (josevalim) [inline lib/phoenix/controller.ex]: ✂️ maybe?

**Comment by chrismccord** [comment]
Scheme is definitely needed. Will make the change, nuke endpoint, and always include /

**Pull request merged**